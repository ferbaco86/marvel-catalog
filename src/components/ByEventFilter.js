import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Select = styled.select`
border: none;
padding-top: 0.5rem;
border-bottom: 3px solid black;
`;

const ByEventFilter = props => {
  const { filter } = props;
  const EVENTS = ['All', 'Secret Wars', 'Civil War', 'Spider-Verse', 'Onslaught', 'Mutant Massacre', 'House of M', 'Avengers VS X-Men', 'Secret Invasion', 'Fear Itself', 'Infinity'];
  return (
    <Select onChange={e => filter(e)}>
      {EVENTS.map(event => (
        <option key={event}>
          {event}
        </option>
      ))}
    </Select>

  );
};

ByEventFilter.propTypes = {
  filter: PropTypes.func.isRequired,
};

export default ByEventFilter;
