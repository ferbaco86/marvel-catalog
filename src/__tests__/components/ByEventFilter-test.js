import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import ByEventFilter from '../../components/ByEventFilter';

const mockFunc = () => 'test';

describe('Rendering component', () => {
  it('creates a ByEventFilter component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <ByEventFilter filter={mockFunc} />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Spider-Verse');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ByEventFilter filter={mockFunc} />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
