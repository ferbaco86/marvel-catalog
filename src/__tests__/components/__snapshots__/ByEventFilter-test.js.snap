// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<select
  className="sc-bdfBwQ jKvFWK"
  onChange={[Function]}
>
  <option>
    All
  </option>
  <option>
    Secret Wars
  </option>
  <option>
    Civil War
  </option>
  <option>
    Spider-Verse
  </option>
  <option>
    Onslaught
  </option>
  <option>
    Mutant Massacre
  </option>
  <option>
    House of M
  </option>
  <option>
    Avengers VS X-Men
  </option>
  <option>
    Secret Invasion
  </option>
  <option>
    Fear Itself
  </option>
  <option>
    Infinity
  </option>
  <option>
    Secret Empire
  </option>
  <option>
    Original Sin
  </option>
</select>
`;
